src/Tools/_framework/Paths/PortfolioActivityEditor.jsx

```diff
@@ -1689,23 +1712,156 @@ export function PortfolioActivityEditor() {
+
+                      <Box bg="doenet.mainGray" h="32px" w="100%">
+                        <HStack
+                          ml="0px"
+                          // ml="33px"
+                          h="32px"
+                          bg="doenet.mainGray"
+                          // bg="doenet.canvas"
+                          pl="10px"
+                          pt="1px"
+                        >
+                          <Popover offset={[119, 5]}>
+                            <PopoverTrigger>
+                              <Tag
+                                data-test="Warning Button"
+                                tabIndex="0"
+                                cursor="pointer"
+                                size="md"
+                                colorScheme={
+                                  errorsAndWarnings.warnings.length == 0
+                                    ? "blackAlpha"
+                                    : "yellow"
+                                }
+                              >
+                                {errorsAndWarnings.warnings.length} Warning
+                                {errorsAndWarnings.warnings.length != 1 && "s"}
+                              </Tag>
+                            </PopoverTrigger>
+                            {errorsAndWarnings.warnings.length == 0 ? (
+                              <PopoverContent data-test="Warning Content">
+                                <PopoverHeader fontWeight="semibold">
+                                  No Warnings
+                                </PopoverHeader>
+                              </PopoverContent>
+                            ) : (
+                              <PopoverContent data-test="Warning Content">
+                                <PopoverArrow />
+                                <PopoverHeader fontWeight="semibold">
+                                  Warning
+                                  {errorsAndWarnings.warnings.length != 1 &&
+                                    "s"}
+                                </PopoverHeader>
+                                <PopoverBody maxH="40vh" overflow="scroll">
+                                  <List spacing={2}>
+                                    {errorsAndWarnings.warnings.map(
+                                      (warningObj, i) => {
+                                        return (
+                                          <ListItem
+                                            key={i}
+                                            data-test={`Warning ${i}`}
+                                          >
+                                            <ListIcon
+                                              as={MdCheckCircle}
+                                              color="yellow.400"
+                                            />
+                                            Line #
+                                            {warningObj.doenetMLrange.lineBegin}{" "}
```

Comment:
It's possible that lineBegin is undefined.  (I'm working to reduce the cases where this happens.) Should handle this case gracefully.

Reply:
Here's a case where we definitely won't have a lineBegin: when the error/warning comes from an external file loaded in via `copyFromURI`.

Reply:
Sometimes doenetMLrange isn't even defined, so at the least it should be
```warningObj.doenetMLrange?.lineBegin```
so this line doesn't throw an error.

Reply:
Done